### PR TITLE
engine: enable log recycling by default

### LINKED ISSUE #14379: [RaftLogEngine] Open `enable-log-recycle` by default.
## Development Task
In the previous release v6.3, we've made RaftEngine support to recycle obsolete logs by the configuration `enable-log-recycle` for reusing them for the newly `append`, which reduces the write-duration of RaftLogs. But this feature is close by default as its unaligned rewrite will bring unexpected Read IOs.

After plenty of tests, we've proved that this feature truly makes more contributions to WRITE, with little side effects on READ. So, we decide to OPEN this feature by default.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What is changed and how it works?

Issue Number: Close #14379 

What's Changed:

```commit-message
Enable log recycling in `RaftLogEngine` by default
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
 - docs-cn: https://github.com/pingcap/docs-cn/pull/13381
 - docs: https://github.com/pingcap/docs/pull/12869
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Enable log recycling in `RaftLogEngine` by default.
```

### CODE REVIEW

**Comment by LykxSassinator** [comment]
@tabokie PTAL, thx.

**Review by tabokie — COMMENTED** [review]
Needs to update config template.

**Comment by LykxSassinator** [comment]
> Needs to update config template.

fixed.

**Comment by LykxSassinator** [comment]
Ping @tabokie @tonyxuqqi .

**Review by tabokie — APPROVED** [review]

**Review by tonyxuqqi — APPROVED** [review]

**Comment by tabokie** [comment]
test failed.

**Comment by LykxSassinator** [comment]
> test failed.

My fault. Missed it, already fixed it.

File: src/config/mod.rs [hunk 1]
```diff
@@ -1800,7 +1800,13 @@ impl Default for RaftEngineConfig {
     fn default() -> Self {
         Self {
             enable: true,
-            config: RawRaftEngineConfig::default(),
+            config: RawRaftEngineConfig {
```
Comment (tabokie) [inline src/config/mod.rs:1803]: Could you move this to validate instead?

Comment (LykxSassinator) [reply src/config/mod.rs:1803]: If we do this, users cannot manually close `enable-log-recycle` in the configuration file.
IMO, in current stage(not release this modification in the `raft-engine` lib), it's better to do this modification at the initialization of `RaftEngineConfig`.

**Pull request merged**